### PR TITLE
Allow default attributes to propagate to child/sibling interactors

### PR DESCRIPTION
**Pull request by mohameddiaa27**
## Description

* Add test cases to cover expected behavior for context default attributes.
* Ensure that default attributes are correctly accessible in all interactors pending to perform.

## Information

- [ ] Contains Documentation
- [x] Contains Tests
- [ ] Contains Breaking Changes

## Related Issues

- see https://github.com/aaronmallen/activeinteractor/issues/277.

### Changed

- `ActiveInteractor::Context::Attributes#merge!` to set attributes in both `ActiveInteractor::Context::Attributes#attributes` and `ActiveInteractor::Context::Attributes#table`
- `ActiveInteractor::Context::Attributes#initialize` to set attributes in both `ActiveInteractor::Context::Attributes#attributes` and `ActiveInteractor::Context::Attributes#table`

### Fixed

- https://github.com/aaronmallen/activeinteractor/issues/277 Allow default attributes to propagate to sibling/child interactors

### CODE REVIEW

File: lib/active_interactor/context/attributes.rb [hunk 1]
```diff
@@ -132,7 +132,7 @@ def merge!(context)
         copy_flags!(context)
 
         merged_context_attributes(context).each_pair do |key, value|
-          public_send("#{key}=", value) unless value.nil?
+          self[key] = value unless value.nil?
```
Comment (aaronmallen) [inline lib/active_interactor/context/attributes.rb:135]: I'm curious why this works... If we look at https://github.com/aaronmallen/activeinteractor/blob/main/lib/active_interactor/context/attributes.rb#L80 we can see that under the hood we're simply calling `public_send`

**Comment by aaronmallen** [comment]
@mohameddiaa27 can you also rebase this off 1-1-stable and point it to merge into that branch?

File: lib/active_interactor/context/attributes.rb (hunk 1 above)
Comment (mohameddiaa27) [reply lib/active_interactor/context/attributes.rb:135]: Yes I also thought that was not necessary until I saw it the behavior shown in failing specs.

The `public_send` defines attribute with the given key and therefore updates `attributes`

https://github.com/aaronmallen/activeinteractor/blob/c4f6bdd81a8d98b06bb590584dba1cc92ab15d55/lib/active_interactor/context/attributes.rb#L80


While `super` updates the `@table`
https://github.com/aaronmallen/activeinteractor/blob/c4f6bdd81a8d98b06bb590584dba1cc92ab15d55/lib/active_interactor/context/attributes.rb#L82

Comment (aaronmallen) [reply lib/active_interactor/context/attributes.rb:135]: ah that makes sense to me then.  Good catch!

File: spec/integration/a_basic_organizer_spec.rb [hunk 2]
```diff
@@ -224,5 +224,91 @@ def perform
...
+        context 'when [:baz] is nil' do
+          let(:context_attributes) { {} }
+
+          it { is_expected.to have_attributes(baz: 'baz', baz_is_set_at_3: true, baz_is_set_at_4: true) }
+        end
+
+        context 'when [:zoo] is nil' do
+          let(:context_attributes) { {} }
+
+          it { is_expected.to have_attributes(zoo: 'zoo', zoo_is_set_at_3: true, zoo_is_set_at_4: true) }
```
Comment (mohameddiaa27) [inline spec/integration/a_basic_organizer_spec.rb:309]: This is another edge case when default attribute does not propagate as expected to child interactors 🤔

Comment (aaronmallen) [reply spec/integration/a_basic_organizer_spec.rb:309]: Hmmm, can you validate `TestOrganizer.context_class` is `TestOrganizerContext`

Comment (mohameddiaa27) [reply spec/integration/a_basic_organizer_spec.rb:332]: Good point 👍 But is seems that it is correctly set https://github.com/aaronmallen/activeinteractor/pull/267/commits/9fce77657c689831533d4ab306a9c8c6606173b7

**Review by aaronmallen — APPROVED** [review]
LGTM! rebase this off of `1-1-stable` and push up a separate pull request set to merge into `1-1-stable` and I'll get them both merged.

**Pull request merged**